### ISSUE TITLE
Workflow continued to onExit immediately even though pod got retried

### ISSUE THREAD

**Issue by jackiemazy** [argoproj/argo#4108]
## Summary 
**What happened?**
While running a single-step argo workflow on openshift, the pod failed on creating container on first try, then succeeded to retry and created the container in a split of second, but the workflow continued to onExit immediately even though pod got retried.

Nothing (pod yaml, pod events, workflow yaml, ...) has record of this retry other than the workflow controller log.

And the workflow watch logs showed the pod in `ContainerCreating` stage, but the workflow already exited.
```bash
Status:              Failed
Message:             child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored
Conditions:
 Completed           True
Created:             Tue Sep 22 13:35:32 -0400 (14 seconds ago)
Started:             Tue Sep 22 13:35:32 -0400 (14 seconds ago)
Finished:            Tue Sep 22 13:35:45 -0400 (1 second ago)
Duration:            13 seconds
ResourcesDuration:   9s*(1 cpu),7s*(100Mi memory)
Parameters:

STEP                              TEMPLATE                        PODNAME                   DURATION  MESSAGE
 ✖ nrdee-wf-lina                  run-nrdee                                                           child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored
 └---◷ run-nrdee-lina-step        run-nrdee-lina                  nrdee-wf-lina-1997874998  1s        ContainerCreating

 ✔ nrdee-wf-lina.onExit           exit-handler
 └---✔ send-exit-notification(0)  notifier-wft/exit-notification  nrdee-wf-lina-1427007037  10s
```

**What you expected to happen?**
The pod retries should have been captured and picked up by argo to get the workflow proceed, not exit.

## Diagnostics

What version of Argo Workflows are you running?
2.10.1

```yaml
$ oc get wf -o yaml nrdee-wf-lina
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  creationTimestamp: 2020-09-22T17:35:32Z
  generation: 1
  labels:
    workflows.argoproj.io/completed: "true"
    workflows.argoproj.io/phase: Failed
  name: nrdee-wf-lina
  namespace: asias-etl
  resourceVersion: "245893691"
  selfLink: /apis/argoproj.io/v1alpha1/namespaces/asias-etl/workflows/nrdee-wf-lina
  uid: 039ab2e3-fcfa-11ea-ae7c-00505688fd38
spec:
  entrypoint: run-nrdee
  onExit: exit-handler
  serviceAccountName: argo
  templates:
  - arguments: {}
    inputs: {}
    metadata: {}
    name: run-nrdee
    outputs: {}
    steps:
    - - arguments: {}
        name: run-nrdee-lina-step
        template: run-nrdee-lina
  - arguments: {}
    inputs: {}
    metadata: {}
    name: exit-handler
    outputs: {}
    steps:
    - - arguments:
          parameters:
          - name: sender-name
            value: asias-etl-nrdee
          - name: workflow-description
            value: asias-etl-nrdee
        name: send-exit-notification
        templateRef:
          name: notifier-wft
          template: exit-notification
  - arguments: {}
    container:
      image: docker-registry-default.okd.mitre.org/asias-etl/nrdee-lina:latest
      name: nrdee-lina
      resources:
        limits:
          cpu: 100m
          memory: 8G
        requests:
          cpu: 100m
          memory: 8G
      volumeMounts:
      - mountPath: /etc/confd/.container/secrets
        name: lina-secret
    inputs: {}
    metadata: {}
    name: run-nrdee-lina
    outputs: {}
  volumes:
  - name: lina-secret
    secret:
      secretName: app-confd-secrets-nrdee-lina
status:
  conditions:
  - status: "True"
    type: Completed
  finishedAt: 2020-09-22T17:35:45Z
  message: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored
  nodes:
    nrdee-wf-lina:
      children:
      - nrdee-wf-lina-3521326840
      displayName: nrdee-wf-lina
      finishedAt: 2020-09-22T17:35:33Z
      id: nrdee-wf-lina
      message: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored
      name: nrdee-wf-lina
      outboundNodes:
      - nrdee-wf-lina-1997874998
      phase: Failed
      startedAt: 2020-09-22T17:35:32Z
      templateName: run-nrdee
      templateScope: local/nrdee-wf-lina
      type: Steps
    nrdee-wf-lina-61367822:
      boundaryID: nrdee-wf-lina-1560128769
      children:
      - nrdee-wf-lina-1427007037
      displayName: send-exit-notification
      finishedAt: 2020-09-22T17:35:45Z
      id: nrdee-wf-lina-61367822
      inputs:
        parameters:
        - name: sender-name
          value: asias-etl-nrdee
        - name: workflow-description
          value: asias-etl-nrdee
      name: nrdee-wf-lina.onExit[0].send-exit-notification
      outputs:
        exitCode: "0"
      phase: Succeeded
      startedAt: 2020-09-22T17:35:33Z
      templateRef:
        name: notifier-wft
        template: exit-notification
      templateScope: local/nrdee-wf-lina
      type: Retry
    nrdee-wf-lina-1427007037:
      boundaryID: nrdee-wf-lina-1560128769
      displayName: send-exit-notification(0)
      finishedAt: 2020-09-22T17:35:43Z
      hostNodeName: okd-comp9.mitre.org
      id: nrdee-wf-lina-1427007037
      inputs:
        parameters:
        - name: sender-name
          value: asias-etl-nrdee
        - name: workflow-description
          value: asias-etl-nrdee
      name: nrdee-wf-lina.onExit[0].send-exit-notification(0)
      outputs:
        exitCode: "0"
      phase: Succeeded
      resourcesDuration:
        cpu: 9
        memory: 7
      startedAt: 2020-09-22T17:35:33Z
      templateRef:
        name: notifier-wft
        template: exit-notification
      templateScope: local/nrdee-wf-lina
      type: Pod
    nrdee-wf-lina-1560128769:
      children:
      - nrdee-wf-lina-4131149733
      displayName: nrdee-wf-lina.onExit
      finishedAt: 2020-09-22T17:35:45Z
      id: nrdee-wf-lina-1560128769
      name: nrdee-wf-lina.onExit
      outboundNodes:
      - nrdee-wf-lina-1427007037
      phase: Succeeded
      startedAt: 2020-09-22T17:35:33Z
      templateName: exit-handler
      templateScope: local/nrdee-wf-lina
      type: Steps
    nrdee-wf-lina-1997874998:
      boundaryID: nrdee-wf-lina
      displayName: run-nrdee-lina-step
      finishedAt: 2020-09-22T17:35:33Z
      hostNodeName: okd-comp8.mitre.org
      id: nrdee-wf-lina-1997874998
      message: ContainerCreating
      name: nrdee-wf-lina[0].run-nrdee-lina-step
      phase: Pending
      startedAt: 2020-09-22T17:35:32Z
      templateName: run-nrdee-lina
      templateScope: local/nrdee-wf-lina
      type: Pod
    nrdee-wf-lina-3521326840:
      boundaryID: nrdee-wf-lina
      children:
      - nrdee-wf-lina-1997874998
      displayName: '[0]'
      finishedAt: 2020-09-22T17:35:33Z
      id: nrdee-wf-lina-3521326840
      message: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored
      name: nrdee-wf-lina[0]
      phase: Error
      startedAt: 2020-09-22T17:35:32Z
      templateName: run-nrdee
      templateScope: local/nrdee-wf-lina
      type: StepGroup
    nrdee-wf-lina-4131149733:
      boundaryID: nrdee-wf-lina-1560128769
      children:
      - nrdee-wf-lina-61367822
      displayName: '[0]'
      finishedAt: 2020-09-22T17:35:45Z
      id: nrdee-wf-lina-4131149733
      name: nrdee-wf-lina.onExit[0]
      phase: Succeeded
      startedAt: 2020-09-22T17:35:33Z
      templateName: exit-handler
      templateScope: local/nrdee-wf-lina
      type: StepGroup
  phase: Failed
  resourcesDuration:
    cpu: 9
    memory: 7
  startedAt: 2020-09-22T17:35:32Z
  storedTemplates:
    namespaced/notifier-wft/exit-notification:
      arguments: {}
      container:
        args:
        - argo-notify
        - '{{inputs.parameters.workflow-description}}'
        - '{{workflow.name}}'
        - '{{workflow.status}}'
        - '{{workflow.failures}}'
        env:
        - name: NOTIFIER_SUPPRESSIONS_YAML
          value: /etc/notifier/suppressions/appsettings.suppressions.yml
        - name: SENDER_EMAIL
          value: noreply@mitre.org
        - name: SENDER_NAME
          value: '{{inputs.parameters.sender-name}}'
        image: docker-registry-default.okd.mitre.org/asias-etl/notifier:latest
        name: notifier
        resources:
          requests:
            cpu: 100m
            memory: 10M
        volumeMounts:
        - mountPath: /etc/confd/.container/secrets
          name: notifier-secret-volume
        - mountPath: /etc/confd/.container/configMaps
          name: notifier-configmap-volume
        - mountPath: /etc/notifier/suppressions
          name: notifier-suppressions-volume
      inputs:
        parameters:
        - name: sender-name
        - name: workflow-description
      metadata: {}
      name: exit-notification
      outputs: {}
      retryStrategy:
        limit: 3
      volumes:
      - name: notifier-secret-volume
        secret:
          secretName: app-confd-secrets-notifier
      - configMap:
          name: app-confd-configmaps-notifier
        name: notifier-configmap-volume
      - configMap:
          name: app-suppressions-configmaps-notifier
        name: notifier-suppressions-volume
```

```
time="2020-09-22T17:35:32Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Updated phase  -> Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Steps node nrdee-wf-lina initialized Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="StepGroup node nrdee-wf-lina-3521326840 initialized Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Pod node nrdee-wf-lina-1997874998 initialized Pending" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Created pod: nrdee-wf-lina[0].run-nrdee-lina-step (nrdee-wf-lina-1997874998)" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Workflow step group node nrdee-wf-lina-3521326840 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:32Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Running resourceVersion=245893464 workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Updating node nrdee-wf-lina-1997874998 message: ContainerCreating"
time="2020-09-22T17:35:33Z" level=error msg="Mark error node" error="pods \"nrdee-wf-lina-1997874998\" is forbidden: exceeded quota: asias-etl, requested: requests.memory=7914900Ki, used: requests.memory=11429715200, limited: requests.memory=16Gi" namespace=asias-etl nodeName="nrdee-wf-lina[0].run-nrdee-lina-step" workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-1997874998 phase Pending -> Error" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-1997874998 message: pods \"nrdee-wf-lina-1997874998\" is forbidden: exceeded quota: asias-etl, requested: requests.memory=7914900Ki, used: requests.memory=11429715200, limited: requests.memory=16Gi" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-1997874998 finished: 2020-09-22 17:35:33.257749188 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Step group node nrdee-wf-lina-3521326840 deemed errored due to child nrdee-wf-lina[0].run-nrdee-lina-step error: pods \"nrdee-wf-lina-1997874998\" is forbidden: exceeded quota: asias-etl, requested: requests.memory=7914900Ki, used: requests.memory=11429715200, limited: requests.memory=16Gi" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-3521326840 phase Running -> Error" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-3521326840 message: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina-3521326840 finished: 2020-09-22 17:35:33.257814459 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="step group nrdee-wf-lina-3521326840 was unsuccessful: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Outbound nodes of nrdee-wf-lina-1997874998 is [nrdee-wf-lina-1997874998]" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Outbound nodes of nrdee-wf-lina is [nrdee-wf-lina-1997874998]" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina phase Running -> Failed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina message: child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="node nrdee-wf-lina finished: 2020-09-22 17:35:33.257881375 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Checking daemoned children of nrdee-wf-lina" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Steps node nrdee-wf-lina-1560128769 initialized Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="StepGroup node nrdee-wf-lina-4131149733 initialized Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Retry node nrdee-wf-lina-61367822 initialized Running" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Pod node nrdee-wf-lina-1427007037 initialized Pending" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Created pod: nrdee-wf-lina.onExit[0].send-exit-notification(0) (nrdee-wf-lina-1427007037)" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:33Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Running resourceVersion=245893493 workflow=nrdee-wf-lina
time="2020-09-22T17:35:34Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:34Z" level=info msg="Updating node nrdee-wf-lina-1997874998 status Error -> Pending"
time="2020-09-22T17:35:34Z" level=info msg="Updating node nrdee-wf-lina-1997874998 message: ContainerCreating"
time="2020-09-22T17:35:34Z" level=info msg="Updating node nrdee-wf-lina-1427007037 message: ContainerCreating"
time="2020-09-22T17:35:34Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:34Z" level=info msg="Skipped pod nrdee-wf-lina.onExit[0].send-exit-notification(0) (nrdee-wf-lina-1427007037) creation: already exists" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:34Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:34Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Running resourceVersion=245893511 workflow=nrdee-wf-lina
time="2020-09-22T17:35:35Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:35Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:35Z" level=info msg="Skipped pod nrdee-wf-lina.onExit[0].send-exit-notification(0) (nrdee-wf-lina-1427007037) creation: already exists" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:35Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:40Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:40Z" level=info msg="Updating node nrdee-wf-lina-1427007037 status Pending -> Running"
time="2020-09-22T17:35:40Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:40Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:40Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Running resourceVersion=245893611 workflow=nrdee-wf-lina
time="2020-09-22T17:35:41Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:41Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:41Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:43Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:43Z" level=info msg="insignificant pod change" key=asias-etl/nrdee-wf-lina-1427007037
time="2020-09-22T17:35:43Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:43Z" level=info msg="Workflow step group node nrdee-wf-lina-4131149733 not yet completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Processing workflow" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Setting node nrdee-wf-lina-1427007037 outputs"
time="2020-09-22T17:35:45Z" level=info msg="Updating node nrdee-wf-lina-1427007037 status Running -> Succeeded"
time="2020-09-22T17:35:45Z" level=info msg="Running OnExit handler: exit-handler" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-61367822 phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-61367822 finished: 2020-09-22 17:35:45.860819602 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Step group node &NodeStatus{ID:nrdee-wf-lina-4131149733,Name:nrdee-wf-lina.onExit[0],DisplayName:[0],Type:StepGroup,TemplateName:exit-handler,TemplateRef:nil,Phase:Running,BoundaryID:nrdee-wf-lina-1560128769,Message:,StartedAt:2020-09-22 17:35:33 +0000 UTC,FinishedAt:0001-01-01 00:00:00 +0000 UTC,PodIP:,Daemoned:nil,Inputs:nil,Outputs:nil,Children:[nrdee-wf-lina-61367822],OutboundNodes:[],StoredTemplateID:,WorkflowTemplateName:,TemplateScope:local/nrdee-wf-lina,ResourcesDuration:ResourcesDuration{},HostNodeName:,} successful" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-4131149733 phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-4131149733 finished: 2020-09-22 17:35:45.860919477 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Outbound nodes of nrdee-wf-lina-61367822 is [nrdee-wf-lina-1427007037]" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Outbound nodes of nrdee-wf-lina-1560128769 is [nrdee-wf-lina-1427007037]" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-1560128769 phase Running -> Succeeded" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="node nrdee-wf-lina-1560128769 finished: 2020-09-22 17:35:45.86100522 +0000 UTC" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Checking daemoned children of nrdee-wf-lina-1560128769" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Updated phase Running -> Failed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Updated message  -> child 'nrdee-wf-lina[0].run-nrdee-lina-step' errored" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Marking workflow completed" namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Checking daemoned children of " namespace=asias-etl workflow=nrdee-wf-lina
time="2020-09-22T17:35:45Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Failed resourceVersion=245893691 workflow=nrdee-wf-lina
time="2020-09-22T17:35:46Z" level=info msg="nrdee-wf-lina released all acquired locks"
time="2020-09-22T17:35:46Z" level=info msg="Workflow update successful" namespace=asias-etl phase=Failed resourceVersion=245893691 workflow=nrdee-wf-lina
time="2020-09-22T17:35:46Z" level=info msg="Labeled pod asias-etl/nrdee-wf-lina-1427007037 completed"
time="2020-09-22T17:35:47Z" level=info msg="Labeled pod asias-etl/nrdee-wf-lina-1427007037 completed"
```
The key being:
```
time="2020-09-22T17:35:34Z" level=info msg="Updating node nrdee-wf-lina-1997874998 status Error -> Pending"
time="2020-09-22T17:35:34Z" level=info msg="Updating node nrdee-wf-lina-1997874998 message: ContainerCreating"
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by simster7** [comment]
It seems what happened was:

1. Pod errored due to resource quota error: `"Mark error node" error="pods \"nrdee-wf-lina-1997874998\" is forbidden: exceeded quota: asias-etl,`
2. Parent StepGroup got marked as Failed for that reason
3. OnExit handler started running
4. The Pod was then able to execute and got marked as Pending.

I see that you don't have the `resubmitPendingPods`... so it seems that something in your environment restarted a Pod that had phase `Failed` and returned it to phase `Pending`. This seems unexpected to me, as Argo considered phase `Failed` to be final.

Do you know if this is the case or similar?

**Comment by jackiemazy** [comment]
Thanks for looking into this. Yes. it appears that the pod got changed from `Error` to `Pending` by openshift. So if openshift does retry, would argo be able to pick that up?

**Comment by alexec** [comment]
> Thanks for looking into this. Yes. it appears that the pod got changed from Error to Pending by openshift. So if openshift does retry, would argo be able to pick that up?

In v2.11 will log this problem - #3949. 

I'm not sure it is something we logically cater for and I'm not sure if behaviour becomes ... undefined.

**Comment by jackiemazy** [comment]
Thanks! I guess we will see. Do we happen to have an ETA on this?

**Comment by alexec** [comment]
> Do we happen to have an ETA on this?

Sorry, no.

**Comment by alexec** [comment]
@jackiemazy we probably cannot support the scenario where a pod goes back to pending. How would we ever know this will happen? We could never complete a workflow. 

Can you please provide more information about this OpenShift issue? For example, how do you enable this feature?

I'm marking as `wontfix`.
